Crashes when searching buffer symbols (using Solargraph and vim mode)

### Describe the bug / provide steps to reproduce it

Summary:

Application crashes when typing a whitespace as a first character in "search buffer symbols" panel/finder using Solargraph as language server. 

Steps to trigger the problem:

- Default LSP configuration / no LSP configuration in `settings.json` file
- Solargraph gem version: 0.50.0  (included in a `Gemfile` of the project)
- Vim mode enabled
 
1. Open a file with valid ruby code containing several method definitions
2. Enter vim-mode
3. Open buffer symbols finder using vim key sequence: `g s`
4. Start typing a whitespace as a first character

Actual Behavior:

Zed crashes

Expected Behavior:

Same behaviour as using other language servers, for example C++ one.

### Zed Version and System Specs

Zed: v0.170.4 (Zed)
OS: macOS 14.5.0
Memory: 16 GiB
Architecture: aarch64

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.

<details><summary>Zed.log</summary>

<!-- Click below this line and paste or drag-and-drop your log-->
```
2025-01-28T17:53:38.817418+01:00 [ERROR] {
  "thread": "main",
  "payload": "called `Option::unwrap()` on a `None` value",
  "location_data": {
    "file": "/Users/administrator/actions-runner-2/_work/zed/zed/crates/language/src/outline.rs",
    "line": 149
  },
  "backtrace": [
    "backtrace::capture::Backtrace::create::hf48ddd49527a7d0c+9011136",
    "backtrace::capture::Backtrace::new::hc55c4d5f459461ca+9010948",
    "zed::reliability::init_panic_hook::{{closure}}::he2bf0495c656e112+77826652",
    "std::panicking::rust_panic_with_hook::h09e8a656f11e82b2+60260008",
    "std::panicking::begin_panic_handler::{{closure}}::h1230eb3cc91b241c+60257064",
    "std::sys::backtrace::__rust_end_short_backtrace::hc3491307aceda2c2+60249864",
    "_rust_begin_unwind+60256292",
    "core::panicking::panic_fmt::ha4b80a05b9fff47a+86303012",
    "core::panicking::panic::h298549a7412a7069+86303120",
    "core::option::unwrap_failed::hb7af631ec4f78cd6+86302892",
    "async_io::driver::block_on::{{closure}}::ha1a93fc5376e4556+41787784",
    "<outline::OutlineViewDelegate as picker::PickerDelegate>::update_matches::h0b60e268ea19d744+41812648",
    "picker::Picker<D>::update_matches::hf7d0c63b93f08d5c+41725928",
    "gpui::window::ViewContext<V>::subscribe::{{closure}}::hf4726c1668885fb7+41634076",
    "gpui::app::AppContext::flush_effects::h36b2998e8955153a+30615680",
    "gpui::platform::PlatformInputHandler::replace_text_in_range::h041b6ac251c04fdc+31034852",
    "gpui::platform::mac::window::insert_text::h894a8478599fd968+31018872",
    "gpui::platform::mac::window::handle_key_event::h13d4407363486240+31002616",
    "<gpui::platform::mac::platform::MacPlatform as gpui::platform::Platform>::run::h6b8279b4a575f716+30931388",
    "gpui::app::App::run::h1a1836eb8f5ae035+77025844",
    "zed::main::hfe18087c28c8b946+77870780",
    "std::sys::backtrace::__rust_begin_short_backtrace::h0e268d9193fd4a2c+77737172",
    "std::rt::lang_start::{{closure}}::h38d2e9fb82905652+78595348",
    "std::rt::lang_start_internal::hdd117cb81a316264+60191344",
    "_main+77881868"
  ],
  "app_version": "0.170.4",
  "release_channel": "stable",
  "target": "aarch64-apple-darwin",
  "os_name": "macOS",
  "os_version": "14.5.0",
  "architecture": "aarch64",
  "panicked_on": 1738083218817,
  "system_id": "5a401a97-4fb3-42eb-8166-ebfa6940e6e6",
  "installation_id": "98619046-3a05-42c2-9fce-fff537a41440",
  "session_id": "d9dd074a-39e8-4b5d-be2a-afb793b25d95"
}
```
<!-- Click above this line and paste or drag-and-drop your log--></details>

--- a/crates/language/src/outline.rs
+++ b/crates/language/src/outline.rs
@@ -146,7 +146,9 @@ impl<T> Outline<T> {
                 }
             } else {
                 let mut name_ranges = outline_match.name_ranges.iter();
-                let mut name_range = name_ranges.next().unwrap();
+                let Some(mut name_range) = name_ranges.next() else {
+                    continue;
+                };
                 let mut preceding_ranges_len = 0;
                 for position in &mut string_match.positions {
                     while *position >= preceding_ranges_len + name_range.len() {
@@ -194,6 +196,40 @@ impl<T> Outline<T> {
 #[cfg(test)]
 mod tests {
     use super::*;
+    use gpui::TestAppContext;
+
+    #[gpui::test]
+    async fn test_entries_with_no_names(cx: &mut TestAppContext) {
+        let outline = Outline::new(vec![
+            OutlineItem {
+                depth: 0,
+                range: Point::new(0, 0)..Point::new(5, 0),
+                text: "class Foo".to_string(),
+                highlight_ranges: vec![],
+                name_ranges: vec![6..9],
+                body_range: None,
+                annotation_range: None,
+            },
+            OutlineItem {
+                depth: 0,
+                range: Point::new(2, 0)..Point::new(2, 7),
+                text: "private".to_string(),
+                highlight_ranges: vec![],
+                name_ranges: vec![],
+                body_range: None,
+                annotation_range: None,
+            },
+        ]);
+        assert_eq!(
+            outline
+                .search(" ", cx.executor())
+                .await
+                .into_iter()
+                .map(|mat| mat.string)
+                .collect::<Vec<String>>(),
+            vec!["class Foo".to_string()]
+        );
+    }
 
     #[test]
     fn test_find_most_similar_with_low_similarity() {

